### PR TITLE
Fixed #4232 App crashes on trying to upload a media from gallery.

### LINKED ISSUE #4232: App crashes on trying to upload a media from gallery.
**Summary:** 
On selecting an item from the gallery for upload the app crashes unexpectedly.

**Steps to reproduce:** 
Try selecting a media from the gallery for upload from the floating action button.

What did you expect the app to do, and what did you see instead?
The app should not crash.

**System logs:**

```
2021-02-16 02:25:03.775 20600-20600/fr.free.nrw.commons.beta E/ACRA: ACRA caught a NullPointerException for fr.free.nrw.commons.beta
    java.lang.NullPointerException: Attempt to invoke virtual method 'long java.lang.Long.longValue()' on a null object reference
        at fr.free.nrw.commons.filepicker.UploadableFile.getDateTimeFromExif(UploadableFile.java:127)
        at fr.free.nrw.commons.filepicker.UploadableFile.getFileCreatedDate(UploadableFile.java:83)
        at fr.free.nrw.commons.upload.UploadModel.createAndAddUploadItem(UploadModel.java:95)
        at fr.free.nrw.commons.upload.UploadModel.preProcessImage(UploadModel.java:84)
        at fr.free.nrw.commons.repository.UploadRepository.preProcessImage(UploadRepository.java:184)
        at fr.free.nrw.commons.upload.mediaDetails.UploadMediaPresenter.receiveImage(UploadMediaPresenter.java:83)
        at fr.free.nrw.commons.upload.mediaDetails.UploadMediaDetailFragment.init(UploadMediaDetailFragment.java:128)
        at fr.free.nrw.commons.upload.mediaDetails.UploadMediaDetailFragment.onViewCreated(UploadMediaDetailFragment.java:114)

```

**Device and Android version:** 

Nokia 6.1+ android 10 stock 
 
**Commons app version:** 
betaDebug  master

**Would you like to work on the issue?**
Yes.

### PR DESCRIPTION
**Pull request by 4D17Y4**
**Description (required)**

Fixes #4232 

What changes did you make and why?

~As @nicolas-raoul said https://github.com/commons-app/apps-android-commons/issues/4232#issuecomment-779903798 this was due to image with no EXIF.
Fixed by removing Exif and using file functions to get last modified and created time.~

Exif interface now returns null and not -1 so changed the check conditions.

**Tests performed (required)**

3.0.0-debug-master~2695bbaf4 on Nokia 6.1 stock android version 10.

### CODE REVIEW

**Comment by 4D17Y4** [comment]
up for review @neslihanturan , @nicolas-raoul , @ashishkumar468

**Review by nicolas-raoul — CHANGES_REQUESTED** [review]
Thanks for the effort, but we really need to parse the EXIF, as file are sometimes touched, edited, moved, transferred from a device to another. I believe it should not be too hard to see what ExifInterface is outputting and adapting to that :-)

**Comment by 4D17Y4** [comment]
Thanks, @nicolas-raoul
I was heading in the wrong direction.
I've made the requested changes.

File: app/src/main/java/fr/free/nrw/commons/filepicker/UploadableFile.java [hunk 1]
```diff
@@ -124,10 +124,10 @@ private DateTimeWithSource getFileCreatedDateFromCP(Context context) {
     private DateTimeWithSource getDateTimeFromExif() {
         try {
             ExifInterface exif = new ExifInterface(file.getAbsolutePath());
-            @SuppressLint("RestrictedApi") long dateTime = exif.getDateTime();
-            if (dateTime != -1) {
-                Date date = new Date(dateTime);
-                return new DateTimeWithSource(date, DateTimeWithSource.EXIF_SOURCE);
+                @SuppressLint("RestrictedApi") Long dateTime = exif.getDateTime();
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/filepicker/UploadableFile.java:127]: Great! Thanks for finding this!
Last thing, would you mind fixing the indentation?

**Review by nicolas-raoul — CHANGES_REQUESTED** [review]

**Comment by nicolas-raoul** [comment]
When uploading an image without EXIF:
- Latest master: crash
- This PR: success

File: app/src/main/java/fr/free/nrw/commons/filepicker/UploadableFile.java (hunk 1 above)
Comment (4D17Y4) [reply app/src/main/java/fr/free/nrw/commons/filepicker/UploadableFile.java]: Fixed.

**Pull request merged**